src/main/java/kr/codesqaud/cafe/controller/UserQueryController.java

```diff
@@ -0,0 +1,77 @@
+package kr.codesqaud.cafe.controller;
+
+import java.util.List;
+
+import javax.servlet.http.HttpSession;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+
+import kr.codesqaud.cafe.domain.User;
+import kr.codesqaud.cafe.service.UserService;
+
+@Controller
+public class UserQueryController {
+	private final UserService userService;
+
+	public UserQueryController(UserService userService) {
+		this.userService = userService;
+	}
+
+	@GetMapping("/user/form")
+	public String getSignUpForm() {
+		return "user/form";
+	}
+
+	@GetMapping("/user/login")
+	public String login() {
+		return "user/login";
+	}
+
+	@GetMapping("/checkForUpdate")
+	public String checkForUpdate() {
+		return "user/checkForUpdate";
+	}
+
+	@GetMapping("/check/{userID}")
+	public String checkUserID(@PathVariable String userID, HttpSession session) {
+		Object value = session.getAttribute("sessionUser");
+		if (value == null) {
+			return "redirect:/user/login";
+		}
+		User user = (User)value;
+		if (!user.getUserID().equals(userID)) {
+			throw new IllegalStateException("다른 사용자의 정보는 수정할 수 없습니다.");
```

Comment:
디미터 법칙을 조사해보시는건 어떠세요? 도메인 자체에서 해당 validation을 할 수 있습니다.
```suggestion
user.validateUserId(userID);
```
User 객체 내부에서 검증하면 컨트롤러에서 처리하지 않고 validate 메서드를 호출만 하면 되겠죠?

Reply:
네! 예외 처리 부분은 좀 더 생각하고 코드를 짜야겠습니다ㅠㅠ 변경했습니당